pkg/util/validation/exporter.go

```diff
@@ -0,0 +1,48 @@
+package validation
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// OverridesExporter exposes per-tenant resource limit overrides as Prometheus metrics
+type OverridesExporter struct {
+	limitSupplier func() map[string]*Limits
+	description   *prometheus.Desc
+}
+
+// NewOverridesExporter creates an OverridesExporter that reads updates to per-tenant
+// limits using the provided function.
+func NewOverridesExporter(limitSupplier func() map[string]*Limits) *OverridesExporter {
+	return &OverridesExporter{
+		limitSupplier: limitSupplier,
+		description: prometheus.NewDesc(
+			"cortex_overrides",
+			"Resource limit overrides applied to tenants",
+			[]string{"limit_name", "user"},
+			nil,
+		),
+	}
+}
+
+func (oe *OverridesExporter) Describe(ch chan<- *prometheus.Desc) {
+	ch <- oe.description
+}
+
+func (oe *OverridesExporter) Collect(ch chan<- prometheus.Metric) {
+	allLimits := oe.limitSupplier()
+	if allLimits == nil {
+		return
+	}
+
+	for tenant, limits := range allLimits {
```

Comment:
Plenty more limits we could return here. Are those not interesting?

Reply:
The ones here are what the`cortex-tools` exporter was making available and seem to be ingester focused. I figure we'd start with these since these seem to be the limits that Grafana Cloud users bump into the most. I can add others now if you have some in mind